src/bin/pgcopydb/ld_transform.c

```diff
@@ -2360,20 +2367,44 @@ stream_write_update(FILE *out, LogicalMessageUpdate *update)
 			return false;
 		}
 
-		uint32_t hash = hashlittle(buf->data, buf->len, 5381);
+		/*
+		 * When all column values in the SET clause are equal to those in the
+		 * WHERE clause, we remove all columns from the SET clause. This results
+		 * in an invalid UPDATE statement like the one shown below:
+		 *
+		 * UPDATE table SET WHERE "id" = 1;
+		 *
+		 * Usually, the above could happen when REPLICA IDENTITY is set to FULL,
+		 * and the UPDATE statement executed with the same values as the old ones.
+		 * For e.g.
+		 * UPDATE table SET "id" = 1 WHERE "id" = 1;
+		 *
+		 * Skip the UPDATE statement in such cases.
+		 */
+		bool skipUpdate = (skippedColumns == new->attributes.count);
 
-		FFORMAT(out, "PREPARE %x AS %s;\n", hash, buf->data);
+		if (skipUpdate)
+		{
+			log_warn("Skipping UPDATE statement as all columns are "
+					 "the same as the old");
+		}
```

Comment:
If we would `return true;` from here, we wouldn't need an else branch at all. What do you think?